ga4gh/datamodel/genotype_phenotype.py

```diff
@@ -17,47 +16,101 @@
 import ga4gh.datamodel as datamodel
 
 
-class AbstractG2PDataset(datamodel.DatamodelObject):
+class AbstractPhenotypeAssociationSet(datamodel.DatamodelObject):
+    compoundIdClass = datamodel.PhenotypeAssociationSetCompoundId
 
-    def __init__(self, setName=None):
-        self._searchQuery = ""
+    def __init__(self, parentContainer, localId):
+        super(AbstractPhenotypeAssociationSet, self).__init__(
+            parentContainer, localId)
 
     def toProtocolElement(self):
-        # TODO remove this... added to follow pattern of datadriven tests
+        pas = protocol.PhenotypeAssociationSet()
+        pas.name = self.getLocalId()
+        pas.id = self.getId()
+        pas.datasetId = self.getParentContainer().getId()
+        pas.info = self.getInfo()
+        return pas
+
+    def getInfo(self):
+        return {}
+
+    def _generateFeature(self, featureId="", childIds=[], parentId="",
```

Comment:
Can you use hanging indents here (and everywhere else) please? I.e.,
```python
def _generateFeature(
    self, featureId="", childIds=[], parentId="", featureSetId="", referenceName="", 
    start=None, end=None, strand=None, _term="", _termId="", sourceName="", 
    sourceVersion=""):
```
I know I'm nagging about these style issues all the time and it's all subjective in any case, but this code objectively 'looks different' to the rest of the code. This is bad for several reasons, but mainly because scripts written/regexps created to do stuff to the code in bulk will assume particular things. If we have different styles for doing things, these will break or at least be a lot harder to do. This point particularly and especially applies to visual indents, because code transformations that we plan on undertaking _will_ break this. If you try to automatically snake_case identifiers with visual indentation you will end up with a complete mess. 

So, using visual indents is actively creating work for the future.If you want we can have a discussion about style choices elsewhere and discuss an update to the project-wide coding conventions. However, until that happens please try and keep any new code that is added more in-line with the style used everywhere else.